app/src/main/java/com/everest/movieapp/data/api/ApplicationContextProvider.kt

```diff
@@ -0,0 +1,20 @@
+package com.everest.movieapp.data.api
+
+import android.app.Application
+import android.content.Context
+
+class ApplicationContextProvider : Application() {
+    companion object {
+        private var instance: ApplicationContextProvider? = null
```

Comment:
Normally putting context at static instance level is discouraged, Can you please check how we can use instance without need of companion object